compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -85,21 +89,22 @@ impl<'a> Parser<'a> {
             self.mk_stmt(lo, StmtKind::Empty)
         } else if self.token != token::CloseDelim(token::Brace) {
             // Remainder are line-expr stmts.
-            let e = self.parse_expr_res(Restrictions::STMT_EXPR, Some(attrs))?;
+            let e = if force_collect == ForceCollect::Yes {
+                self.collect_tokens_no_attrs(|this| {
```

Comment:
This should be a call to `collect_tokens_trailing_token`, with `attrs` passed in (since we do have outer attributes).

Reply:
I assumed this case is analogous to `parse_expr_force_collect` (https://github.com/rust-lang/rust/pull/84995/files#r629329546) so we can use `collect_tokens_no_attrs`.

Also, I don't think we can simply pass `attrs` here because they don't necessary apply to the whole statement, they can apply only to some sub-expression like in `#[attr] 1 + 2` (grouped as `(#[attr] 1) + 2`).